src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolver.cs

```diff
@@ -0,0 +1,189 @@
+                    }
+                }
+            }
+        }
+
+        internal override string GetModuleName(DkmClrModuleInstance module)
+        {
+            return module.Name;
+        }
+
+        internal override MetadataReader GetModuleMetadata(DkmClrModuleInstance module)
+        {
+            uint length;
+            IntPtr ptr;
+            try
+            {
+                ptr = module.GetMetaDataBytesPtr(out length);
+            }
+            catch (Exception e) when (IsBadOrMissingMetadataException(e))
+            {
+                return null;
+            }
+            Debug.Assert(length > 0);
+            unsafe
+            {
+                return new MetadataReader((byte*)ptr, (int)length);
+            }
+        }
+
+        internal override DkmRuntimeFunctionResolutionRequest[] GetRequests(DkmProcess process)
+        {
+            return process.GetRuntimeFunctionResolutionRequests();
+        }
+
+        internal override string GetRequestModuleName(DkmRuntimeFunctionResolutionRequest request)
+        {
+            return request.ModuleName;
+        }
+
+        internal override void OnFunctionResolved(
+            DkmClrModuleInstance module,
+            DkmRuntimeFunctionResolutionRequest request,
+            int token,
+            int version,
+            int ilOffset)
+        {
+            var address = DkmClrInstructionAddress.Create(
+                module.RuntimeInstance,
+                module,
+                new DkmClrMethodId(Token: token, Version: 1),
+                NativeOffset: uint.MaxValue,
+                ILOffset: 0,
+                CPUInstruction: null);
+            request.OnFunctionResolved(address);
+        }
+
+        private static readonly Guid s_messageSourceId = new Guid("ac353c9b-c599-427b-9424-cbe1ad19f81e");
+
+        private static bool ShouldEnable(DkmProcess process)
+        {
+            var message = DkmCustomMessage.Create(
+                process.Connection,
+                process,
+                s_messageSourceId,
+                MessageCode: 1, // Should enable this resolver?
+                Parameter1: null,
+                Parameter2: null);
+            try
+            {
+                var reply = message.SendLower();
+                var result = (int)reply.Parameter1;
+                // Possible values are 0 = false, 1 = true, 2 = not ready.
+                // At this point, we should only get 0 or 1, but to be
+                // safe, treat any value other than 1 as false.
+                Debug.Assert(result == 0 || result == 1);
+                return result == 1; 
+            }
+            catch (NotImplementedException)
+            {
+                return false;
+            }
+        }
+
+        private const uint COR_E_BADIMAGEFORMAT = 0x8007000b;
+        private const uint CORDBG_E_MISSING_METADATA = 0x80131c35;
+
+        private static bool IsBadOrMissingMetadataException(Exception e)
```

Comment:
You may have already looked at this and decided it's not worth it, but it seems like it might be nice to link in ```MetadataUtilities.cs``` (or refactor it into partial classes so that part of it could be linked in)...seems like there is at least *some* duplicated code around MetadataReader helpers...